### DISCUSSION TITLE
Suggestion to get rid of the "stopTime" storage property in the pro contract

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/evm-monorepo#221] [Ideas]
The `stopTime` property redundant because the same information can be gleaned from the last segment milestone.

The only place where it is needed is the [`getWithdrawableAmount`](https://github.com/sablierhq/v2-core/blob/aeb6f4c1ff2102753cd898c015b7480aa9b9bff2/src/SablierV2Pro.sol#L135). It might be slightly more gas efficient to keep it as is in the `getWithdrawableAmount`, but this comes at a cost of 20,000 gas for committing this property in storage and, more importantly, at the cost of making it difficult to avoid the [StackTooDeep issue](https://soliditydeveloper.com/stacktoodeep).

Finally, I should clarify that this proposal is strictly about the `stopTime` storage property - the `getStopTime` getter is still helpful, and I am happy to keep it.

**Comment by andreivladbrg** [comment]
> but this comes at a cost of 20,000 gas for committing this property in storage and

That's wrong, it would barely affect the gas. It would fit in a non complete storage slot

> at the cost of making it difficult to avoid the [StackTooDeep issue](https://soliditydeveloper.com/stacktoodeep).

Was it difficult to avoid it in the `create` or in the `getWithdrawableAmount` function?

**Reply by PaulRBerg** [reply]
> It would fit in a non complete storage slot

Good point!

> Was it difficult to avoid it in the `create` or `getWithdrawableAmount` function?

In `create`. But I'm open to adding the `stopTime` back if it works now.

**Reply by andreivladbrg** [reply]
Haven't tested, but my gut is telling me that it should work.

**Comment by PaulRBerg** [comment]
Locking, we ended up bringing this property back (#240).
